From: Enrico Jorns <ejo@pengutronix.de>
Date: Tue, 13 Jan 2015 13:21:29 +0100
Subject: [PATCH] v4l2: changed min_buffers to min_buffers_for_capture

---
 sys/v4l2/gstv4l2videoenc.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sys/v4l2/gstv4l2videoenc.c b/sys/v4l2/gstv4l2videoenc.c
index e3cbd29df199..4987c77f29a1 100644
--- a/sys/v4l2/gstv4l2videoenc.c
+++ b/sys/v4l2/gstv4l2videoenc.c
@@ -582,7 +582,8 @@ gst_v4l2_video_enc_decide_allocation (GstVideoEncoder *
     ret =
         GST_VIDEO_ENCODER_CLASS
         (parent_class)->decide_allocation (encoder, query);
-  latency = self->v4l2capture->min_buffers * self->v4l2capture->duration;
+  latency =
+      self->v4l2capture->min_buffers_for_capture * self->v4l2capture->duration;
   gst_video_encoder_set_latency (encoder, latency, latency);
   return ret;
 }
