From: Peter Seiderer <ps.report@gmx.net>
Date: Fri, 12 Dec 2014 14:47:04 +0100
Subject: [PATCH] v4l2videoenc: send V4L2_ENC_CMD_STOP instead of empty buffers

Signed-off-by: Peter Seiderer <ps.report@gmx.net>
---
 sys/v4l2/gstv4l2videoenc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sys/v4l2/gstv4l2videoenc.c b/sys/v4l2/gstv4l2videoenc.c
index 519dff096680..abd2ca4cfbd3 100644
--- a/sys/v4l2/gstv4l2videoenc.c
+++ b/sys/v4l2/gstv4l2videoenc.c
@@ -741,6 +741,10 @@ gst_v4l2_video_enc_sink_event (GstVideoEncoder * encoder, GstEvent * event)
       gst_v4l2_object_unlock (self->v4l2output);
       gst_v4l2_object_unlock (self->v4l2capture);
       break;
+    case GST_EVENT_EOS:
+      GST_DEBUG_OBJECT (self, "got EOS event, send V4L2_ENC_CMD_STOP");
+      gst_v4l2_encoder_cmd (self->v4l2output, V4L2_ENC_CMD_STOP, 0);
+      break;
     default:
       break;
   }
