From: Michael Grzeschik <m.grzeschik@pengutronix.de>
Date: Thu, 21 May 2015 15:21:29 +0200
Subject: [PATCH] gstv4l2videoenc: ignore interlace mode

Signed-off-by: Michael Grzeschik <m.grzeschik@pengutronix.de>
---
 sys/v4l2/gstv4l2videoenc.c | 51 ++++++++++++++++++++++++++++++++++++++++------
 1 file changed, 45 insertions(+), 6 deletions(-)

diff --git a/sys/v4l2/gstv4l2videoenc.c b/sys/v4l2/gstv4l2videoenc.c
index da92cf93193c..92f13d811fb0 100644
--- a/sys/v4l2/gstv4l2videoenc.c
+++ b/sys/v4l2/gstv4l2videoenc.c
@@ -699,6 +699,41 @@ gst_v4l2_video_enc_src_query (GstVideoEncoder * encoder, GstQuery * query)
   return ret;
 }
 
+/* copies the given caps */
+static GstCaps *
+gst_v4l2_transform_caps_remove_interlace_mode_info (GstCaps * caps)
+{
+  GstStructure *st;
+  GstCapsFeatures *f;
+  gint i, n;
+  GstCaps *res;
+
+  res = gst_caps_new_empty ();
+
+  n = gst_caps_get_size (caps);
+  for (i = 0; i < n; i++) {
+    st = gst_caps_get_structure (caps, i);
+    f = gst_caps_get_features (caps, i);
+
+    /* If this is already expressed by the existing caps
+     * skip this structure */
+    if (i > 0 && gst_caps_is_subset_structure_full (res, st, f))
+      continue;
+
+    st = gst_structure_copy (st);
+    /* Only remove format info for the cases when we can actually convert */
+    if (!gst_caps_features_is_any (f)
+        && gst_caps_features_is_equal (f,
+            GST_CAPS_FEATURES_MEMORY_SYSTEM_MEMORY)) {
+      gst_structure_remove_fields (st, "interlace-mode", NULL);
+    }
+
+    gst_caps_append_structure_full (res, st, gst_caps_features_copy (f));
+  }
+
+  return res;
+}
+
 static gboolean
 gst_v4l2_video_enc_sink_query (GstVideoEncoder * encoder, GstQuery * query)
 {
@@ -708,6 +743,7 @@ gst_v4l2_video_enc_sink_query (GstVideoEncoder * encoder, GstQuery * query)
   switch (GST_QUERY_TYPE (query)) {
     case GST_QUERY_CAPS:{
       GstCaps *filter, *result = NULL;
+      GstCaps *tmp, *tmp2;
       GstPad *pad = GST_VIDEO_ENCODER_SINK_PAD (encoder);
 
       gst_query_parse_caps (query, &filter);
@@ -717,17 +753,20 @@ gst_v4l2_video_enc_sink_query (GstVideoEncoder * encoder, GstQuery * query)
       else
         result = gst_pad_get_pad_template_caps (pad);
 
+      tmp = gst_v4l2_transform_caps_remove_interlace_mode_info (result);
+
       if (filter) {
-        GstCaps *tmp = result;
-        result =
-            gst_caps_intersect_full (filter, tmp, GST_CAPS_INTERSECT_FIRST);
+        tmp2 = gst_caps_intersect_full (filter, tmp, GST_CAPS_INTERSECT_FIRST);
         gst_caps_unref (tmp);
+        tmp = tmp2;
       }
 
-      GST_DEBUG_OBJECT (self, "Returning sink caps %" GST_PTR_FORMAT, result);
-
-      gst_query_set_caps_result (query, result);
+      GST_DEBUG_OBJECT (self, "Returning sinky lllcaps %" GST_PTR_FORMAT,
+          result);
       gst_caps_unref (result);
+
+      gst_query_set_caps_result (query, tmp);
+      gst_caps_unref (tmp);
       break;
     }
 
