From: Philipp Zabel <p.zabel@pengutronix.de>
Date: Fri, 5 Feb 2016 13:41:22 +0100
Subject: [PATCH] reset: Share struct reset_control between reset_control_get
 calls

Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
---
 drivers/reset/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/reset/core.c b/drivers/reset/core.c
index df4486d38d84..957750600ff3 100644
--- a/drivers/reset/core.c
+++ b/drivers/reset/core.c
@@ -224,7 +224,7 @@ struct reset_control *__of_reset_control_get(struct device_node *node,
 	}
 
 	if (WARN_ON(args.args_count != rcdev->of_reset_n_cells)) {
-		mutex_unlock(&reset_controller_list_mutex);
+		mutex_unlock(&reset_list_mutex);
 		return ERR_PTR(-EINVAL);
 	}
 
